packages/config-types/src/ExpoConfig.ts

```diff
@@ -3,6 +3,11 @@
  * The standard Expo config object defined in `app.config.js` files.
  */
 
+/**
+ * Specifies the JavaScript engine for apps. Supported only on EAS Build. Defaults to `jsc`. Valid values: `hermes`, `jsc`.
+ */
+export type JsEngine = 'hermes' | 'jsc';
```

Comment:
This file is generated, first you need to add the types to expo/universe (example: https://github.com/expo/universe/pull/7876) then you can run:

```
yarn generate --path ../../../universe/server/www/xdl-schemas/UNVERSIONED-schema.json
```
[reference](https://github.com/expo/expo-cli/blob/master/packages/config-types/README.md#contributing)